### PR TITLE
Loader hoc

### PR DESCRIPTION
**Pull request by DavidRojas1612**
## Summary
-Completed use loader hoc in the fetching matches data.

## Screenshots
image

## Trello Card
https://trello.com/c/Sv1UzOvP

### CODE REVIEW

File: react/tic-tac-toe/src/app/components/WithLoader/styles.module.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@import 'src/scss/variables';
+.container {
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/components/WithLoader/styles.module.scss]: Add an empty line here

File: react/tic-tac-toe/src/app/components/WithLoader/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import Spinner from 'react-spinkit';
+
+import styles from './styles.module.scss';
+
+function withLoader(WrappedComponent) {
+  return function Loader({ loader, ...props }) {
+    if (loader) {
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/components/WithLoader/index.js]:
```
return loader ? (
        <div className={styles.container}>
          <Spinner name="double-bounce" fadeIn="none" />
        </div>
      ) : (
      <WrappedComponent {...props} />
     );
```

File: react/tic-tac-toe/src/app/screens/Home/screens/Matches/layout.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+              </h3>
+            </li>
+          ))}
+      </ol>
+    </div>
+  );
+}
+
+LayoutWithLoader.prototype = {
+  matches: objectOf(matchPropType)
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/screens/Home/screens/Matches/layout.js]: `objectOf` ?? what about `shape` ?

**Review by alejobh — COMMENTED** [review]
Nice work, BUT the corrections you did about redux-recompose should go to this PR https://github.com/wolox-training/dr-react/pull/18/ not here  🤔

File: react/tic-tac-toe/src/app/components/WithLoader/index.js (hunk 2 above)
Comment (jmtov) [inline react/tic-tac-toe/src/app/components/WithLoader/index.js]: I'd suggest this prop is named `loading` instead. It's a bit clearer. What do you think?

File: react/tic-tac-toe/src/app/screens/Home/screens/Matches/index.js [hunk 4]
```diff
@@ -1,43 +1,18 @@
 import React, { useEffect } from 'react';
-import Spinner from 'react-spinkit';
-import { useSelector, useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 
 import gameActions from '~redux/game/actions';
 
-import styles from './styles.module.scss';
+import LayoutWithLoader from './layout';
```
Comment (jmtov) [inline react/tic-tac-toe/src/app/screens/Home/screens/Matches/index.js]: I'd name it just `Layout`. Although, what we usually do here at Wolox is:
`index.js => MatchesContainer`
`layout.js => Matches`

**Review by jmtov — CHANGES_REQUESTED** [review]
Nice and simple. I like it! I left a couple of comments, please check them out! 😺

File: react/tic-tac-toe/src/app/components/WithLoader/index.js (hunk 2 above)
Comment (DavidRojas1612) [reply react/tic-tac-toe/src/app/components/WithLoader/index.js]: In my opinion I think it is not necessary to use a ternary operator.

File: react/tic-tac-toe/src/redux/game/reducer.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { completeReducer, createReducer, completeState } from 'redux-recompose';
+import { completeReducer, createReducer, completeState, onReadValue } from 'redux-recompose';
```
Comment (alejobh) [inline react/tic-tac-toe/src/redux/game/reducer.js:1]: Why is this `onReadValue` here?, check the status of your base branch and this branch :)

**Review by alejobh — APPROVED** [review]

**Review by jmtov — APPROVED** [review]
Nice!

**Pull request merged**